olive/strategy/search_parameter.py

```diff
@@ -130,21 +160,19 @@ def __init__(
         self.support = support
         self.default = default or self.get_invalid_choice()
 
-    def get_support(self) -> List[Any]:
-        raise NotImplementedError("Use get_support_with_args instead")
+    def get_parents(self) -> Tuple[str]:
+        return self.parents
 
-    def get_support_with_args(
-        self, parent_values: Dict[str, Any]
-    ) -> Union[List[str], List[int], List[float], List[bool]]:
+    def get_support(self, parent_values: Dict[str, Any] = None) -> Union[List[str], List[int], List[float], List[bool]]:
         """Get the support for the search parameter for a given parent value."""
-        # pylint: disable=arguments-differ
-        assert parent_values.keys() == set(self.parents), "parent values keys do not match the parents"
+        assert parent_values is not None
+        assert set(self.parents).issubset(parent_values.keys()), "Not all parents included in parent values keys"
         parent_values = tuple(parent_values[parent] for parent in self.parents)
         return self.support.get(parent_values, self.default).get_support()
 
     def condition(self, parent_values: Dict[str, Any]) -> SearchParameter:
         """Fix the parent value and return a new search parameter."""
-        assert set(parent_values.keys()).issubset(set(self.parents)), "parent values keys not a subset of the parents"
+        assert set(self.parents).issubset(parent_values.keys()), "Not all parents includes in parent values keys"
```

Comment:
This check makes `condition` the same as `get_support`. In condition, only a subset of the original parents can be provided.

Reply:
> In condition, only a subset of the original parents can be provided.

Not sure what you mean by "only a subset of the original parents". What's _original parents_? I don't see `self.parents` changing in this context. 

Just so I understand your proposal, are you proposing the following -
```
parent_values = {k: v for k, v in parent_values.items() if k in self.parents}
assert set(parent_values.keys()).issubset(set(self.parents))
```
This assertion would never ever be false and will never fire. Limiting `parent_values` to keys in `parents` and then asserting that `parent_values` has only such keys?

`parent_values` is always a superset because it can contain values for potentially all search parameters in the search space. `parents` is, however, is a subset of that same set. That's exactly what the code is asserting.

Reply:
True, the assert is not needed. The original assert was to ensure at least one of the parents is present in `parent_values` (which previously did not include non parents). But I missed the fact that an empty set is also a subset.
An equivalent now would be replacing the old assert with
```
parent_values = {k: v for k,v in parent_values.items() if k in self.parents}
if not parent_values:
    return self
```

The `set(self.parents).issubset(parent_values.keys())` is not what we want. It asserts that all parents are present in `parent_values` but condition doesn't need that.

Reply:
> which previously did not include non parents.

It still doesn't include non-parents. The `parent_values` is built using whatever the `get_parents` returns which is not any different from previous implementation.

> It asserts that all parents are present in parent_values but condition doesn't need that.

Hmm is that true? Wouldn't all parents be present in `parent_values`, even if one's value is default?

Regardless, this is not the intended change for the PR itself so I will revert it back to whatever it was.